### ISSUE TITLE
Default methods on managed model interfaces cannot be called on Java 9

### ISSUE THREAD

**Issue by eriwen** [gradle/gradle#721]
Related to https://issues.apache.org/jira/browse/GROOVY-7879 we seem to do something wrong when generating the proxies for the `@Managed` model. The test `InterfaceBackedManagedTypeIntegrationTest` fails on Java 9 with

```
java.lang.IncompatibleClassChangeError: Method Person.getName()Ljava/lang/String; must be InterfaceMethodref constant
```

**Comment by lpedrosa** [comment]
I believe this is a groovy bug. The groovy team seemed to have fixed something when calling static methods, which are defined on interfaces:

* https://issues.apache.org/jira/browse/GROOVY-7879
* https://issues.apache.org/jira/browse/GROOVY-8164

However, something has changed on Java 9. I have created a [simple example](https://github.com/lpedrosa/groovy-static-interface) that reproduces the error shown above.

Other projects seems to be having the same problem:

* https://github.com/palantir/gradle-docker/issues/96
* The jshell gradle plugin I am trying to contribute to

Not sure what is the best approach here. Should this be raised as an issue in the Groovy project github?

**Comment by sormuras** [comment]
According to the state of the linked Groovy issue 7879 it should be resolved since Groovy 2.4.11.

But, using
```
------------------------------------------------------------
Gradle 4.6
------------------------------------------------------------

Build time:   2018-02-28 13:36:36 UTC
Revision:     8fa6ce7945b640e6168488e4417f9bb96e4ab46c

Groovy:       2.4.12
Ant:          Apache Ant(TM) version 1.9.9 compiled on February 2 2017
JVM:          10-ea ("Oracle Corporation" 10-ea+42)
OS:           Windows 10 10.0 amd64

R:\dev\github\junit-team\junit5>gradlew build
```

with this Java 9 code:
```
java.util.spi.ToolProvider.findFirst('jar').get()
```

...it still fails with `Method java.util.spi.ToolProvider.findFirst(Ljava/lang/String;)Ljava/util/Optional; must be InterfaceMethodref constant`

**Comment by sormuras** [comment]
Happens with Gradle 4.7 (Groovy 2.4.12) as well.
Seems to be fixed with Groovy 3.0-alpha-2, though:
```
groovy "java.util.spi.ToolProvider.findFirst('jar').get().run(System.out, System.err, '--version')"
WARNING: Using incubator modules: jdk.incubator.httpclient

WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.codehaus.groovy.vmplugin.v7.Java7$1 (file:/../lib/groovy-3.0.0-alpha-2.jar) to constructor java.lang.invoke.MethodHandles$Lookup(java.lang.Class,int)
WARNING: Please consider reporting this to the maintainers of org.codehaus.groovy.vmplugin.v7.Java7$1
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release

jar 10
```

**Comment by sormuras** [comment]
Seems to be fixed in 2.4.16: https://issues.apache.org/jira/browse/GROOVY-8528

**Comment by oehme** [comment]
This works fine with Groovy 2.4.15
